// Copyright 2023 Intrinsic Innovation LLC

syntax = "proto3";

package xfa.icon.actions.proto;

import "intrinsic/icon/proto/joint_space.proto";
import "intrinsic/kinematics/types/joint_limits.proto";

// Fixed parameters for a point to point move action. These are provided once by
// the user when the action is created, and remain the same for the lifetime of
// the action.
message PointToPointMoveFixedParams {
  // The position to move to.
  intrinsic_proto.icon.JointVec goal_position = 1;

  // The desired velocity when the goal position is reached. Set to the zero
  // vector to stop at the goal position.
  intrinsic_proto.icon.JointVec goal_velocity = 2;

  // The joint limits to apply to the motion.
  // The actual limits used will be the most conservative of (i) Physical
  // hardware limits, (ii) limits configured on the ICON server; and (iii) these
  // optional action-specific limits.
  optional intrinsic_proto.JointLimits joint_limits = 3;
}
